django/db/models/fields/json.py

```diff
@@ -99,6 +99,8 @@ def get_internal_type(self):
         return "JSONField"
 
     def get_db_prep_value(self, value, connection, prepared=False):
+        if not prepared:
+            value = self.get_prep_value(value)
```

Comment:
That's what I imagined at first too, but other fields seems to call it at the beginning (`DateField`, `DecimalField`, ...). Beside there are 3 returns throughout the JSONField.get_db_prep_value`, so it would be heavier to call it at the end.

Reply:
As a follow up on this, I created https://forum.djangoproject.com/t/model-fields-calling-get-prep-value-from-get-db-prep-value/20947

Reply:
We should call `self.get_prep_value()` as `JSONField()` doesn't implement `get_prep_value()` and we don't want to call `Field.get_prep_value()` instead.